src/app/static/src/app/store/adrUpload/actions.ts

```diff
@@ -111,23 +112,23 @@ export const actions: ActionTree<ADRUploadState, RootState> & ADRUploadActions =
                 requestParams["resourceId"] = resourceId
             }
             if (resourceType === rootState.adr.schemas?.outputSummary) {
-                downloadId = rootState.downloadResults.summary.downloadId
+                downloadId = rootState.downloadResults[DownloadType.SUMMARY].downloadId
```

Comment:
What do you also think about handling this through the `downloadConfig`? Have some list of files which need to go to the ADR, and here look through the `DownloadType` and include those which are in the list?

Reply:
Could use it above in `UploadModal`

Reply:
the reason i didnt do this here is because this requires me to refactor the ADR upload stuff because we need some way of matching up if "outputSummary" exists, then include `DownloadType.SUMMARY`, i say we do a smaller refactor in a separate PR and sort out the adrUpload stuff and there we can put these mappings, infact some of the adrUpload stuff can probably just come from the downloadConfig itself

in the `UploadModal` we have this same mapping issue and also have `outputFileError` which has a weird control flow which does something different if both spectrum and summary have metadata errors vs just one of them, i would like to properly investigate that and see if theres a pattern that we can extract

i can create a ticket for these two if you agree!

Reply:
Yep sounds great doing that separately, yes please create the tickets.